internal/command/test_test.go

```diff
@@ -2838,6 +2839,177 @@ There is no backend type named "foobar".
+						Name: "a",
+					}.Instance(addrs.NoKey).Absolute(addrs.RootModuleInstance),
+					&states.ResourceInstanceObjectSrc{
+						AttrsJSON: []byte(`{"id":"12345","value":"outdated-value-from-prior-test-run"}`), // Is updated to match test cases' expectedState
+						Status:    states.ObjectReady,
+					},
+					addrs.AbsProviderConfig{
+						Provider: addrs.NewDefaultProvider("test"),
+						Module:   addrs.RootModule,
+					},
+				)
+			}),
+			expectedState: `test_resource.a:
+  ID = 12345
+  provider = provider["registry.terraform.io/hashicorp/test"]
+  destroy_fail = false
+  value = value-from-run-that-controls-backend`,
+		},
+	}
+
+	for tn, tc := range testCases {
+		t.Run(tn, func(t *testing.T) {
+			// SETUP
+			td := t.TempDir()
+			testCopyDir(t, testFixturePath(path.Join("test", tc.dirName)), td)
+			defer testChdir(t, td)()
+
+			provider := testing_command.NewProvider(nil)
+
+			providerSource, close := newMockProviderSource(t, map[string][]string{
+				"test": {"1.0.0"},
+			})
+			defer close()
+
+			streams, done := terminal.StreamsForTesting(t)
+			view := views.NewView(streams)
+			ui := new(cli.MockUi)
+
+			meta := Meta{
+				testingOverrides: metaOverridesForProvider(provider.Provider),
+				Ui:               ui,
+				View:             view,
+				Streams:          streams,
+				ProviderSource:   providerSource,
+			}
+
+			// SET & ASSERT PRIOR STATE
+			if tc.priorState != nil {
+				setLocalState(t, tc.localStatePath, tc.priorState)
+
+				actualState := retrieveLocalState(t, tc.localStatePath)
+				if diff := cmp.Diff(actualState, tc.priorState.String()); len(diff) > 0 {
+					t.Errorf("prior state didn't match expected:\nexpected:\n%s\nactual:\n%s\ndiff:\n%s", "", actualState, diff)
+				}
+			}
+
+			// INIT
+			init := &InitCommand{
+				Meta: meta,
+			}
+
+			output := done(t)
+			if code := init.Run(nil); code != 0 {
+				t.Fatalf("expected status code 0 but got %d: %s", code, output.All())
+			}
+
+			// TEST
+			// Reset the streams for the next command.
+			streams, done = terminal.StreamsForTesting(t)
+			meta.Streams = streams
+			meta.View = views.NewView(streams)
+
+			c := &TestCommand{
+				Meta: meta,
+			}
+
+			code := c.Run([]string{"-no-color"})
+			output = done(t)
+
+			// ASSERTIONS
+			if code != 0 {
+				t.Errorf("expected status code 0 but got %d", code)
+			}
+			stdErr := output.Stderr()
+			if len(stdErr) > 0 {
+				t.Fatalf("unexpected error output:\n%s", stdErr)
+			}
+
+			actualState := retrieveLocalState(t, tc.localStatePath)
+			if diff := cmp.Diff(actualState, tc.expectedState); len(diff) > 0 {
+				t.Fatalf("state didn't match expected:\nexpected:\n%s\nactual:\n%s\ndiff:\n%s", "", actualState, diff)
+			}
+
+			// TODO - In future this will not be correct- deletion should not include things contained in the state file.
+			if provider.ResourceCount() > 0 {
+				t.Fatalf("should have deleted all resources on completion but left %v", provider.ResourceString())
+			}
```

Comment:
In future I imagine this test would be updated to assert that `provider.ResourceCount() == 1` and `test_resource.a` hasn't been deleted. But that feature would be added in a subsequent PR.